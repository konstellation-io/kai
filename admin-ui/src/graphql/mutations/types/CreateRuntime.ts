/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import {
  CreateRuntimeInput,
  ErrorCode,
  RuntimeStatus,
  VersionStatus
} from './../../types/globalTypes';

// ====================================================
// GraphQL mutation operation: CreateRuntime
// ====================================================

export interface CreateRuntime_createRuntime_errors {
  __typename: 'Error';
  code: ErrorCode;
  message: string;
}

export interface CreateRuntime_createRuntime_runtime_creationAuthor {
  __typename: 'User';
  id: string;
  email: string;
}

export interface CreateRuntime_createRuntime_runtime_publishedVersion {
  __typename: 'Version';
  status: VersionStatus;
}

export interface CreateRuntime_createRuntime_runtime {
  __typename: 'Runtime';
  id: string;
  name: string;
  status: RuntimeStatus;
  creationDate: string;
  creationAuthor: CreateRuntime_createRuntime_runtime_creationAuthor;
  publishedVersion: CreateRuntime_createRuntime_runtime_publishedVersion | null;
}

export interface CreateRuntime_createRuntime {
  __typename: 'CreateRuntimeResponse';
  errors: CreateRuntime_createRuntime_errors[] | null;
  runtime: CreateRuntime_createRuntime_runtime | null;
}

export interface CreateRuntime {
  createRuntime: CreateRuntime_createRuntime;
}

export interface CreateRuntimeVariables {
  input: CreateRuntimeInput;
}
