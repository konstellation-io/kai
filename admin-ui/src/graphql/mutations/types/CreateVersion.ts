/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import {
  CreateVersionInput,
  ErrorCode,
  VersionStatus
} from './../../types/globalTypes';

// ====================================================
// GraphQL mutation operation: CreateVersion
// ====================================================

export interface CreateVersion_createVersion_errors {
  __typename: 'Error';
  code: ErrorCode;
  message: string;
}

export interface CreateVersion_createVersion_version_creationAuthor {
  __typename: 'User';
  id: string;
  email: string;
}

export interface CreateVersion_createVersion_version {
  __typename: 'Version';
  id: string;
  name: string;
  description: string;
  status: VersionStatus;
  creationDate: string;
  creationAuthor: CreateVersion_createVersion_version_creationAuthor;
}

export interface CreateVersion_createVersion {
  __typename: 'CreateVersionResponse';
  errors: CreateVersion_createVersion_errors[] | null;
  version: CreateVersion_createVersion_version;
}

export interface CreateVersion {
  createVersion: CreateVersion_createVersion;
}

export interface CreateVersionVariables {
  input: CreateVersionInput;
}
