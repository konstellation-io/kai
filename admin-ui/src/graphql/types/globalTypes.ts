/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export enum AccessLevel {
  ADMINISTRATOR = 'ADMINISTRATOR',
  MANAGER = 'MANAGER',
  VIEWER = 'VIEWER'
}

export enum ConfigurationVariableType {
  FILE = 'FILE',
  VARIABLE = 'VARIABLE'
}

export enum LogLevel {
  DEBUG = 'DEBUG',
  ERROR = 'ERROR',
  INFO = 'INFO',
  WARN = 'WARN'
}

export enum NodeStatus {
  ERROR = 'ERROR',
  STARTED = 'STARTED',
  STOPPED = 'STOPPED'
}

export enum RuntimeStatus {
  CREATING = 'CREATING',
  ERROR = 'ERROR',
  STARTED = 'STARTED'
}

export enum UserActivityType {
  CREATE_RUNTIME = 'CREATE_RUNTIME',
  CREATE_VERSION = 'CREATE_VERSION',
  LOGIN = 'LOGIN',
  LOGOUT = 'LOGOUT',
  PUBLISH_VERSION = 'PUBLISH_VERSION',
  START_VERSION = 'START_VERSION',
  STOP_VERSION = 'STOP_VERSION',
  UNPUBLISH_VERSION = 'UNPUBLISH_VERSION',
  UPDATE_SETTING = 'UPDATE_SETTING',
  UPDATE_VERSION_CONFIGURATION = 'UPDATE_VERSION_CONFIGURATION'
}

export enum VersionStatus {
  PUBLISHED = 'PUBLISHED',
  STARTED = 'STARTED',
  STARTING = 'STARTING',
  STOPPED = 'STOPPED'
}

export interface ConfigurationVariablesInput {
  key: string;
  value: string;
}

export interface CreateRuntimeInput {
  name: string;
}

export interface CreateVersionInput {
  file: any;
  runtimeId: string;
}

export interface PublishVersionInput {
  versionId: string;
  comment: string;
}

export interface SettingsInput {
  authAllowedDomains?: string[] | null;
  authAllowedEmails?: string[] | null;
  sessionLifetimeInDays?: number | null;
}

export interface StartVersionInput {
  versionId: string;
}

export interface StopVersionInput {
  versionId: string;
}

export interface UnpublishVersionInput {
  versionId: string;
}

export interface UpdateConfigurationInput {
  versionId: string;
  configurationVariables: ConfigurationVariablesInput[];
}

//==============================================================
// END Enums and Input Objects
//==============================================================
