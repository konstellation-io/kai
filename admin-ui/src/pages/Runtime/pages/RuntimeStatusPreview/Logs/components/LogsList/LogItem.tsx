import React, { useState } from 'react';
import moment from 'moment';

import IconInfo from '@material-ui/icons/Info';
import IconExpand from '@material-ui/icons/ArrowDownward';

import { NodeLog, LogLevel } from '../../../../../../../graphql/models';

import styles from './LogsList.module.scss';
import cx from 'classnames';

const LEVEL_TO_ICON: { [key: string]: any } = {
  [LogLevel.INFO]: IconInfo
};

function LogItem({ date, nodeId, podId, message, level }: NodeLog) {
  const [opened, setOpened] = useState<boolean>(false);

  const dateFormatted = moment(date).format('YYYY-MM-DD');
  const hourFormatted = moment(date).format('hh:mm:ss');

  const LevelIcon = LEVEL_TO_ICON[level];

  function toggleOpenStatus() {
    setOpened(!opened);
  }

  return (
    <div
      className={cx(styles.container, {
        [styles.opened]: opened
      })}
    >
      <div className={styles.row1}>
        <div className={styles.icon}>
          <LevelIcon className="icon-small" />
        </div>
        <div className={styles.date}>{dateFormatted}</div>
        <div className={styles.hour}>{hourFormatted}</div>
        {/* <div className={styles.nodeId}>{nodeId}</div> */}
        {/* <div className={styles.podId}>{`[${podId}]`}</div> */}
        <div className={styles.message}>{message}</div>
        <div className={styles.expand} onClick={toggleOpenStatus}>
          <IconExpand className="icon-regular" />
        </div>
      </div>
      {opened && <div className={styles.messageComplete}>{message}</div>}
    </div>
  );
}

export default LogItem;
