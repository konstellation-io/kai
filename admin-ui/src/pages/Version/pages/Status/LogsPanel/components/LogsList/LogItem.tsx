import React, { memo, useState, useRef, RefObject } from 'react';
import moment from 'moment';
import LevelIcon from '../../../../../../../components/LevelIcon/LevelIcon';
import IconExpand from '@material-ui/icons/ArrowDownward';
import IconCopyAll from '@material-ui/icons/FileCopy';
import IconCopyMessage from '@material-ui/icons/Subject';
import styles from './LogsList.module.scss';
import cx from 'classnames';
import { GetServerLogs_logs_items } from '../../../../../../../graphql/queries/types/GetServerLogs';
import ContextMenu, {
  MenuCallToAction
} from '../../../../../../../components/ContextMenu/ContextMenu';

function onCopyToClipboard(input: RefObject<HTMLInputElement>) {
  if (input.current !== null) {
    input.current.select();
    input.current.setSelectionRange(0, 99999);

    document.execCommand('copy');
  }
}

interface Props extends GetServerLogs_logs_items {
  alwaysOpened: boolean;
}
function LogItem({
  date,
  workflowName,
  nodeName,
  message,
  level,
  alwaysOpened
}: Props) {
  const [opened, setOpened] = useState<boolean>(false);
  const dateFormatted = moment(date).format('YYYY-MM-DD');
  const hourFormatted = moment(date).format('HH:mm:ss.SSS');
  const messageRef = useRef<HTMLInputElement>(null);
  const traceRef = useRef<HTMLInputElement>(null);

  function getTrace() {
    return `${level}  [${date}] ${workflowName}:${nodeName} - ${message}`;
  }

  function toggleOpenStatus() {
    setOpened(!opened);
  }

  function copyMessage() {
    onCopyToClipboard(messageRef);
  }
  function copyTrace() {
    onCopyToClipboard(traceRef);
  }

  const contextMenuActions: MenuCallToAction[] = [
    {
      Icon: IconCopyMessage,
      text: 'copy log message',
      callToAction: copyMessage
    },
    {
      Icon: IconCopyAll,
      text: 'copy complete log trace',
      callToAction: copyTrace
    }
  ];

  return (
    <ContextMenu actions={contextMenuActions}>
      <div>
        <input
          type="text"
          value={message}
          ref={messageRef}
          readOnly
          className={styles.clipboardInput}
        />
        <input
          type="text"
          value={getTrace()}
          ref={traceRef}
          readOnly
          className={styles.clipboardInput}
        />
        <div
          className={cx(styles.container, styles[level], {
            [styles.opened]: opened || alwaysOpened
          })}
        >
          <div className={cx(styles.row1, styles[level])}>
            <div className={styles.icon}>
              <LevelIcon level={level} />
            </div>
            <div className={styles.date}>{dateFormatted}</div>
            <div className={styles.hour}>{hourFormatted}</div>
            <div className={styles.workflow} title={workflowName || ''}>
              {workflowName || ''}
            </div>
            <div className={styles.node} title={nodeName || ''}>
              {nodeName}
            </div>
            <div className={styles.message}>{message}</div>
            {!alwaysOpened && (
              <div className={styles.expand} onClick={toggleOpenStatus}>
                <IconExpand className="icon-regular" />
              </div>
            )}
          </div>
          {(alwaysOpened || opened) && (
            <div className={styles.messageComplete}>{message}</div>
          )}
        </div>
      </div>
    </ContextMenu>
  );
}

export default memo(LogItem);
