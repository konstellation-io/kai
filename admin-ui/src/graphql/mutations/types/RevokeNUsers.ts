/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { UsersInput, AccessLevel } from './../../types/globalTypes';

// ====================================================
// GraphQL mutation operation: RevokeNUsers
// ====================================================

export interface RevokeNUsers_revokeUsers {
  __typename: 'User';
  id: string;
  email: string;
  accessLevel: AccessLevel;
  dateAdded: string;
  lastAccess: string;
}

export interface RevokeNUsers {
  revokeUsers: RevokeNUsers_revokeUsers[];
}

export interface RevokeNUsersVariables {
  input: UsersInput;
}
