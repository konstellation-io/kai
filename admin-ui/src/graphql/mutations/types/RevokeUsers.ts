/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { UsersInput, AccessLevel } from './../../types/globalTypes';

// ====================================================
// GraphQL mutation operation: RevokeUsers
// ====================================================

export interface RevokeUsers_revokeUsers {
  __typename: 'User';
  id: string;
  email: string;
  accessLevel: AccessLevel;
  dateAdded: string;
  lastAccess: string;
}

export interface RevokeUsers {
  revokeUsers: RevokeUsers_revokeUsers[];
}

export interface RevokeUsersVariables {
  input: UsersInput;
}
