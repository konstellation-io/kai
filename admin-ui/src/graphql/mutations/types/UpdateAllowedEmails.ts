/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { SettingsInput } from './../../types/globalTypes';

// ====================================================
// GraphQL mutation operation: UpdateAllowedEmails
// ====================================================

export interface UpdateAllowedEmails_updateSettings {
  __typename: 'Settings';
  authAllowedEmails: string[];
}

export interface UpdateAllowedEmails {
  updateSettings: UpdateAllowedEmails_updateSettings;
}

export interface UpdateAllowedEmailsVariables {
  input: SettingsInput;
}
