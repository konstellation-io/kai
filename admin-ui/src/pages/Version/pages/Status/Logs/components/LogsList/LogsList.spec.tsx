import React from 'react';
import LogItem from './LogItem';
import IconExpand from '@material-ui/icons/ArrowDownward';
import { shallow } from 'enzyme';
import { LogLevel } from '../../../../../../../graphql/types/globalTypes';

describe('LogItem', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <LogItem
        __typename="NodeLog"
        level={LogLevel.DEBUG}
        nodeName="nodeName"
        date={'2020-01-01'}
        message={'some message'}
      />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('show right components', () => {
    expect(wrapper.exists('.container')).toBeTruthy();
    expect(wrapper.find('.message').text()).toBe('some message');
  });

  it('closed by default', () => {
    expect(wrapper.exists('.opened')).toBeFalsy();
  });

  it('opens on open log click', () => {
    wrapper
      .find(IconExpand)
      .parent()
      .simulate('click');
    expect(wrapper.exists('.opened')).toBeTruthy();
    expect(wrapper.find('.messageComplete').text()).toBe('some message');
  });
});

const mockData = {
  data: {
    logs: [
      {
        nodeId: 'someNode1',
        message: 'some message 1',
        date: '2020-01-01'
      },
      {
        nodeId: 'someNode2',
        message: 'some message 2',
        date: '2020-01-01'
      }
    ]
  },
  refetch: jest.fn(),
  fetchMore: jest.fn(),
  subscribeToMore: jest.fn()
};
jest.mock('react-router', () => ({
  useParams: jest.fn(() => ({
    runtimeId: 'someId'
  }))
}));
const mockWriteData = jest.fn();
jest.mock('@apollo/react-hooks', () => ({
  useQuery: jest.fn(() => mockData),
  useSubscription: jest.fn(() => mockData),
  useApolloClient: () => ({
    writeData: mockWriteData
  })
}));
