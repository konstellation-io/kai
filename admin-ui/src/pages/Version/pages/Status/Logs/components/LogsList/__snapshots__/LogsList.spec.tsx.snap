// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogItem matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row1"
  >
    <div
      className="icon"
    >
      <InfoIcon
        className="icon-small"
      />
    </div>
    <div
      className="date"
    >
      2020-01-01
    </div>
    <div
      className="hour"
    >
      00:00:00
    </div>
    <div
      className="message"
    >
      some message
    </div>
    <div
      className="expand"
      onClick={[Function]}
    >
      <ArrowDownwardIcon
        className="icon-regular"
      />
    </div>
  </div>
</div>
`;

exports[`LogsList matches snapshot 1`] = `
<LogsList
  nodeId="nodeId"
  onUpdate={[MockFunction]}
  runtimeId="runtimeId"
>
  <div
    className="listContainer"
    id="VersionLogsListContainer"
  >
    <LogItem
      date="2020-01-01"
      message="some message 1"
      nodeId="someNode1"
    >
      <div
        className="container"
      >
        <div
          className="row1"
        >
          <div
            className="icon"
          >
            <ForwardRef
              className="icon-small"
            >
              <WithStyles(ForwardRef(SvgIcon))
                className="icon-small"
              >
                <ForwardRef(SvgIcon)
                  className="icon-small"
                  classes={
                    Object {
                      "colorAction": "MuiSvgIcon-colorAction",
                      "colorDisabled": "MuiSvgIcon-colorDisabled",
                      "colorError": "MuiSvgIcon-colorError",
                      "colorPrimary": "MuiSvgIcon-colorPrimary",
                      "colorSecondary": "MuiSvgIcon-colorSecondary",
                      "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                      "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                      "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                      "root": "MuiSvgIcon-root",
                    }
                  }
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root icon-small"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                    />
                  </svg>
                </ForwardRef(SvgIcon)>
              </WithStyles(ForwardRef(SvgIcon))>
            </ForwardRef>
          </div>
          <div
            className="date"
          >
            2020-01-01
          </div>
          <div
            className="hour"
          >
            00:00:00
          </div>
          <div
            className="message"
          >
            some message 1
          </div>
          <div
            className="expand"
            onClick={[Function]}
          >
            <ForwardRef
              className="icon-regular"
            >
              <WithStyles(ForwardRef(SvgIcon))
                className="icon-regular"
              >
                <ForwardRef(SvgIcon)
                  className="icon-regular"
                  classes={
                    Object {
                      "colorAction": "MuiSvgIcon-colorAction",
                      "colorDisabled": "MuiSvgIcon-colorDisabled",
                      "colorError": "MuiSvgIcon-colorError",
                      "colorPrimary": "MuiSvgIcon-colorPrimary",
                      "colorSecondary": "MuiSvgIcon-colorSecondary",
                      "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                      "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                      "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                      "root": "MuiSvgIcon-root",
                    }
                  }
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root icon-regular"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </ForwardRef(SvgIcon)>
              </WithStyles(ForwardRef(SvgIcon))>
            </ForwardRef>
          </div>
        </div>
      </div>
    </LogItem>
    <LogItem
      date="2020-01-01"
      message="some message 2"
      nodeId="someNode2"
    >
      <div
        className="container"
      >
        <div
          className="row1"
        >
          <div
            className="icon"
          >
            <ForwardRef
              className="icon-small"
            >
              <WithStyles(ForwardRef(SvgIcon))
                className="icon-small"
              >
                <ForwardRef(SvgIcon)
                  className="icon-small"
                  classes={
                    Object {
                      "colorAction": "MuiSvgIcon-colorAction",
                      "colorDisabled": "MuiSvgIcon-colorDisabled",
                      "colorError": "MuiSvgIcon-colorError",
                      "colorPrimary": "MuiSvgIcon-colorPrimary",
                      "colorSecondary": "MuiSvgIcon-colorSecondary",
                      "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                      "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                      "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                      "root": "MuiSvgIcon-root",
                    }
                  }
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root icon-small"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                    />
                  </svg>
                </ForwardRef(SvgIcon)>
              </WithStyles(ForwardRef(SvgIcon))>
            </ForwardRef>
          </div>
          <div
            className="date"
          >
            2020-01-01
          </div>
          <div
            className="hour"
          >
            00:00:00
          </div>
          <div
            className="message"
          >
            some message 2
          </div>
          <div
            className="expand"
            onClick={[Function]}
          >
            <ForwardRef
              className="icon-regular"
            >
              <WithStyles(ForwardRef(SvgIcon))
                className="icon-regular"
              >
                <ForwardRef(SvgIcon)
                  className="icon-regular"
                  classes={
                    Object {
                      "colorAction": "MuiSvgIcon-colorAction",
                      "colorDisabled": "MuiSvgIcon-colorDisabled",
                      "colorError": "MuiSvgIcon-colorError",
                      "colorPrimary": "MuiSvgIcon-colorPrimary",
                      "colorSecondary": "MuiSvgIcon-colorSecondary",
                      "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                      "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                      "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                      "root": "MuiSvgIcon-root",
                    }
                  }
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root icon-regular"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </ForwardRef(SvgIcon)>
              </WithStyles(ForwardRef(SvgIcon))>
            </ForwardRef>
          </div>
        </div>
      </div>
    </LogItem>
  </div>
</LogsList>
`;
